rust/agents/relayer/src/msg/metadata/base.rs

```diff
@@ -99,6 +207,82 @@ impl BaseMetadataBuilder {
         }
     }
 
+    #[instrument(err, skip(self), fields(destination_domain=self.destination_domain().name()))]
```

Comment:
It's because the MetadataBuilders now require a `MessageMetadataBuilder` instead of a `BaseMetadataBuilder` -- so `let cloned = self.clone_with_incremented_depth()?;` that's called in here does the cloning & incrementing of `MessageMetadataBuilder`